### PR TITLE
Reformat where it's sensible (PEP8), Add coding=utf-8 comments

### PR DESCRIPTION
**Pull request by knockoutMice**
Hi,

I reformatted the docstrings / some long code lines, so it follows the Python style standard (<79).
Now I can split my editor and still read (almost) everything without horizontal scrolling, also PyCharm doesn't cry about it anymore. I kept some long lines unchanged, because they were more readable that way.

Also fixed some typos and added utf-8 comments.

Btw I enjoy using your library, keep up the good work! Although I've got some concerns about the validity of the tests, because at first glance it doesn't seem to separate the features, so the tests are only valid when checked in an interaction-only scenario. I'd be glad to work out some solution / open an issue if that's the case. Of course: correct me if I'm wrong.

Best regards,
Mice Pápai

### CODE REVIEW

File: lightfm/lightfm.py [hunk 1]
```diff
@@ -770,18 +783,18 @@ def get_params(self, deep=True):
...
-                  'learning_rate': self.learning_rate,
-                  'k': self.k,
-                  'n': self.n,
-                  'rho': self.rho,
-                  'epsilon': self.epsilon,
-                  'max_sampled': self.max_sampled,
-                  'item_alpha': self.item_alpha,
-                  'user_alpha': self.user_alpha,
-                  'random_state': self.random_state}
+                  'no_components':     self.no_components,
```
Comment (maciejkula) [inline lightfm/lightfm.py]: Could we leave this change out? I prefer not aligning things.

**Comment by maciejkula** [comment]
This looks great, thanks a lot!

I keep to (and enforce in the lints) a 100-column line width, but these changes look good anyway.

You could, if you like, add yourself to the contributors file as well.

Can you explain what you mean about validity of tests?

File: lightfm/evaluation.py [hunk 2]
```diff
@@ -135,8 +138,9 @@ def auc_score(model, test_interactions, train_interactions=None,
               user_features=None, item_features=None,
               preserve_rows=False, num_threads=1):
     """
-    Measure the ROC AUC metric for a model: the probability that a randomly chosen positive
-    example has a higher score than a randomly chosen negative example.
+    Measure the ROC AUC metric for a model: the probability that a randomly
```
Comment (maciejkula) [inline lightfm/evaluation.py]: Can you have a look at the CircleCI build results and address the lint violations?

**Comment by knockoutMice** [comment]
Made the requested changes. 

When I tried to follow the method you outlined in the readme to run the tests (`python setup.py test`) it didn't work. [I included the output + the pip freeze output.](https://github.com/lyst/lightfm/files/980968/lightfm_python_setuppy_test.txt) I could make it work by calling `.venv36/bin/py.test -xv tests`. Of course my venv was active the whole time.

About the evaluation: the items in the test and train dataset differ by the interactions, but the features remain the same for both, right? My main concern is that measurements only evaluate the prediction of interactions and not the prediction of features.

**Comment by maciejkula** [comment]
Not entirely sure what you mean by your comments re: tests. But the changes look good, thanks!

**Pull request merged**